In [1]:
import pandas as pd
import numpy as np

1. Identificação do Jogo:
  * Name - Nome do Jogo
  * Platform - Plataforma em que o jogo foi lançada
  * Year_of_Release - Ano do lançamento
  * Genre - Genero do Jogo
  * Publisher - Quem esta publicando o jogo

2. Vendas:
  * NA_Sales - Venda na América do Norte
  * EU_Sales - Vendas na União Europeia
  * JP_Sales - Venda no Japão
  * Other_Sales - Soma  das vendas em outros locais
  * Global_Sales - Soma das vendas globais

3. Avaliações:
  * Critic_Score - Nota dos Críticos
  * Critic_Count - Quantidade de críticos que votaram
  * User_Score - Notas dos Usuários
  * User_Count - Quantidade de usuários que votaram

4. Informações Adicionais:
  * Developer - Quem desenvolveu o jogo
  * Rating - Classificação etária do jogo

In [2]:
video_games=pd.read_csv('./data/Video_Games.csv')

In [3]:
video_games.head(5)

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN

In [4]:
# Resumo estatístico para variáveis numéricas
video_games.describe(include=[np.number])


Year_of_Release      NA_Sales      EU_Sales      JP_Sales  \
count     16450.000000  16719.000000  16719.000000  16719.000000   
mean       2006.487356      0.263330      0.145025      0.077602   
std           5.878995      0.813514      0.503283      0.308818   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2020.000000     41.360000     28.960000     10.220000   

        Other_Sales  Global_Sales  Critic_Score  Critic_Count    User_Count  
count  16719.000000  16719.000000   8137.000000   8137.000000   7590.000000  
mean       0.047332      0.533543     68.967679     26.360821    162.229908  
std        0.186710      1.547935     13.938165     18.980495    561.282326  
min        0.000000      0.010000     13.000000      3.000000      4.000000  
25%        0.000000      0.060000     60.000000     12.000000     10.000000  
50%        0.010000      0.170000     71.000000     21.000000     24.000000  
75%        0.030000      0.470000     79.000000     36.000000     81.000000  
max       10.570000     82.530000     98.000000    113.000000  10665.000000

In [5]:
video_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       10015 non-null  object 
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


In [6]:
video_games.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

Identificamos as seguintes colunas com valores nulos:

* Colunas com poucos valores ausentes:
  * Name: 2 valores nulos
  * Genre: 2 valores nulos
  * Publisher: 54 valores nulos
  * Year_of_Release: 269 valores nulos

* Colunas com uma quantidade significativa de valores ausentes:
  * Critic_Score e Critic_Count: 8.582 valores nulos
  * User_Score e User_Count: 6.704 e 9.129 valores nulos, respectivamente
  * Developer: 6.623 valores nulos
  * Rating: 6.769 valores nulos

In [7]:
video_games[video_games['Name'].isnull()]

Name Platform  Year_of_Release Genre              Publisher  NA_Sales  \
659    NaN      GEN           1993.0   NaN  Acclaim Entertainment      1.78   
14246  NaN      GEN           1993.0   NaN  Acclaim Entertainment      0.00   

       EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
659        0.53      0.00         0.08          2.39           NaN   
14246      0.00      0.03         0.00          0.03           NaN   

       Critic_Count User_Score  User_Count Developer Rating  
659             NaN        NaN         NaN       NaN    NaN  
14246           NaN        NaN         NaN       NaN    NaN

In [8]:
# removendo as linhas com valores nulos na coluna Name e Genre, são as mesmas linhas, são apenas duas e essenciais.
video_games = video_games.drop(659)
video_games = video_games.drop(14246)

In [9]:
video_games.isnull().sum()

Name                  0
Platform              0
Year_of_Release     269
Genre                 0
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8580
Critic_Count       8580
User_Score         6702
User_Count         9127
Developer          6621
Rating             6767
dtype: int64

In [10]:
# Preenche valores ausentes com a moda na coluna Rating
mode_value = video_games['Rating'].mode()[0]
video_games.fillna({'Rating': mode_value}, inplace=True)

In [11]:
# Cria o dicionário de mapeamento dos valores categóricos para valores discretos
rating_mapping = {
    'E': 0,    # Everyone
    'M': 1,    # Mature
    'T': 2,    # Teen
    'E10+': 3, # Everyone 10+
    'K-A': 4,  # Kids to Adults
    'EC': 5,   # Early Childhood
    'RP': 6,    # Rating Pending
    'AO': 7    # Adults Only
}
# Aplica a conversão da coluna 'Rating' para valores discretos
video_games['Rating'] = video_games['Rating'].map(rating_mapping)

In [12]:
video_games.isnull().sum()

Name                  0
Platform              0
Year_of_Release     269
Genre                 0
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8580
Critic_Count       8580
User_Score         6702
User_Count         9127
Developer          6621
Rating                0
dtype: int64

In [13]:
# Remove as colunas que contém diversos faltantes e que não são interessante para a análise de vendas
video_games = video_games.drop(['Critic_Score', 'Critic_Count', 'User_Score', 'User_Count'], axis=1)

In [14]:
video_games.isnull().sum()

Name                  0
Platform              0
Year_of_Release     269
Genre                 0
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Developer          6621
Rating                0
dtype: int64

In [15]:
video_games[video_games['Publisher'].isnull()]

Name Platform  \
475                           wwe Smackdown vs. Raw 2006      PS2   
1301                                      Triple Play 99       PS   
1667        Shrek / Shrek 2 2-in-1 Gameboy Advance Video      GBA   
2212                                  Bentley's Hackpack      GBA   
3148   Nicktoons Collection: Game Boy Advance Video V...      GBA   
3159   SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
3755   SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
4127                                  Sonic the Hedgehog      PS3   
4508   The Fairly Odd Parents: Game Boy Advance Video...      GBA   
4615   The Fairly Odd Parents: Game Boy Advance Video...      GBA   
5294       Dragon Ball Z: Budokai Tenkaichi 2 (JP sales)      Wii   
5633   Cartoon Network Collection: Game Boy Advance V...      GBA   
6256   The Legend of Zelda: The Minish Cap(weekly JP ...      GBA   
6415            Sonic X: Game Boy Advance Video Volume 1      GBA   
6532   Dora the Explorer: Game Boy Advance Video Volu...      GBA   
6626   Cartoon Network Collection: Game Boy Advance V...      GBA   
6831      All Grown Up!: Game Boy Advance Video Volume 1      GBA   
7189   Nicktoons Collection: Game Boy Advance Video V...      GBA   
7333                      Yu Yu Hakusho: Dark Tournament      PS2   
7477   SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
7930                    Thomas the Tank Engine & Friends      GBA   
8317     Dragon Ball GT: Game Boy Advance Video Volume 1      GBA   
8328   Codename: Kids Next Door: Game Boy Advance Vid...      GBA   
8358   Teenage Mutant Ninja Turtles: Game Boy Advance...      GBA   
8494                                        Stronghold 3       PC   
8754   Cartoon Network Collection: Game Boy Advance V...      GBA   
8837   Pokémon: Johto Photo Finish: Game Boy Advance ...      GBA   
8883   Strawberry Shortcake: Game Boy Advance Video V...      GBA   
9511                              Farming Simulator 2011       PC   
9739   Super Robot Wars OG Saga: Masou Kishin II - Re...      PSP   
10378                   Disney Channel Collection Vol. 1      GBA   
10487             Atsumare! Power Pro Kun no DS Koushien       DS   
11093                       Action Man-Operation Extreme       PS   
11546  Cartoon Network Collection: Game Boy Advance V...      GBA   
12525                               Chou Soujuu Mecha MG       DS   
12569       Prinny: Can I Really Be The Hero? (US sales)      PSP   
13318                     Monster Hunter Frontier Online      PS3   
13712                          B.L.U.E.: Legend of Water       PS   
14128                                       Housekeeping       DS   
14352                                Bikkuriman Daijiten       DS   
14687                                     World of Tanks     X360   
14772                                 UK Truck Simulator       PC   
15024  Umineko no Naku Koro ni San: Shinjitsu to Gens...      PS3   
15340                                       Mario Tennis      3DS   
15417  Nicktoons Collection: Game Boy Advance Video V...      GBA   
15442                   Demolition Company: Gold Edition       PC   
15680  Moshi, Kono Sekai ni Kami-sama ga Iru to suru ...      PSV   
16019                                       Dream Dancer       DS   
16280                    Homeworld Remastered Collection       PC   
16296             AKB1/48: Idol to Guam de Koishitara...     X360   
16306  Super Robot Monkey Team: Game Boy Advance Vide...      GBA   
16332                        Brothers in Arms: Furious 4     X360   
16339                                  Dance with Devils      PSV   
16662                             Driving Simulator 2011       PC   

       Year_of_Release         Genre Publisher  NA_Sales  EU_Sales  JP_Sales  \
475                NaN      Fighting       NaN      1.57      1.02      0.00   
1301               NaN        Sports       NaN      0.81      0.55      0.00   
1667            

Pesquisando sobre os jogos com publisher null

In [16]:
# https://pt.wikipedia.org/wiki/WWE_SmackDown_vs.*_Raw_2006
video_games.loc[475]

Name               wwe Smackdown vs. Raw 2006
Platform                                  PS2
Year_of_Release                           NaN
Genre                                Fighting
Publisher                                 NaN
NA_Sales                                 1.57
EU_Sales                                 1.02
JP_Sales                                  0.0
Other_Sales                              0.41
Global_Sales                              3.0
Developer                                 NaN
Rating                                      0
Name: 475, dtype: object

In [17]:
# Corrigindo informações da linha 475
video_games.loc[475, 'Publisher'] = 'HQ JP'
video_games.loc[475, 'Developer'] = "Yuke's"
video_games.loc[475, 'Year_of_Release'] = 2006.0

In [18]:
# https://en.wikipedia.org/wiki/Triple_Play_99
video_games.loc[1301]

Name               Triple Play 99
Platform                       PS
Year_of_Release               NaN
Genre                      Sports
Publisher                     NaN
NA_Sales                     0.81
EU_Sales                     0.55
JP_Sales                      0.0
Other_Sales                   0.1
Global_Sales                 1.46
Developer                     NaN
Rating                          0
Name: 1301, dtype: object

In [19]:
# Corrigindo informações da linha 1301
video_games.loc[1301, 'Publisher'] = 'EA Sports'
video_games.loc[1301, 'Developer'] = "EA Canada"
video_games.loc[1301, 'Year_of_Release'] = 1998.0

In [20]:
# https://gamefaqs.gamespot.com/gba/934475-game-boy-advance-video-shrek-*2*
video_games.loc[1667]

Name               Shrek / Shrek 2 2-in-1 Gameboy Advance Video
Platform                                                    GBA
Year_of_Release                                          2007.0
Genre                                                      Misc
Publisher                                                   NaN
NA_Sales                                                   0.87
EU_Sales                                                   0.32
JP_Sales                                                    0.0
Other_Sales                                                0.02
Global_Sales                                               1.21
Developer                                                   NaN
Rating                                                        0
Name: 1667, dtype: object

In [21]:
# Corrigindo informações da linha 1667
video_games.loc[1667, 'Publisher'] = 'Majesco'
video_games.loc[1667, 'Developer'] = "Majesco"


In [22]:
# https://rugrats.fandom.com/wiki/Nicktoons_Collection:_Volume_1
video_games.loc[3148]

Name               Nicktoons Collection: Game Boy Advance Video V...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.46
EU_Sales                                                        0.17
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.64
Developer                                                        NaN
Rating                                                             0
Name: 3148, dtype: object

In [23]:
# Corrigindo informações da linha 3148
video_games.loc[3148, 'Publisher'] = 'Nintendo'
video_games.loc[3148, 'Developer'] = 'Majesco'

In [24]:
# https://gba-video.fandom.com/wiki/SpongeBob_SquarePants_Volume_1
video_games.loc[3159]

Name               SpongeBob SquarePants: Game Boy Advance Video ...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.46
EU_Sales                                                        0.17
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.64
Developer                                                        NaN
Rating                                                             0
Name: 3159, dtype: object

In [25]:
# Corrigindo informações da linha 3159
video_games.loc[3159, 'Publisher'] = 'Nintendo'
video_games.loc[3159, 'Developer'] = 'Majesco'

In [26]:
# https://gba-video.fandom.com/wiki/SpongeBob_SquarePants_Volume_2
video_games.loc[3755]

Name               SpongeBob SquarePants: Game Boy Advance Video ...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.38
EU_Sales                                                        0.14
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.53
Developer                                                        NaN
Rating                                                             0
Name: 3755, dtype: object

In [27]:
# Corrigindo informações da linha 3755
video_games.loc[3755, 'Publisher'] = 'Nintendo'
video_games.loc[3755, 'Developer'] = 'Majesco'

In [28]:
# https://pt.wikipedia.org/wiki/Sonic_the_Hedgehog_(jogo_eletr%C3%B4nico_de_2006)
video_games.loc[4127]

Name               Sonic the Hedgehog
Platform                          PS3
Year_of_Release                   NaN
Genre                        Platform
Publisher                         NaN
NA_Sales                          0.0
EU_Sales                         0.48
JP_Sales                          0.0
Other_Sales                       0.0
Global_Sales                     0.48
Developer                  Sonic Team
Rating                              3
Name: 4127, dtype: object

In [29]:
# Corrigindo informações da linha 4127
video_games.loc[4121, 'Publisher'] = 'Sega'
video_games.loc[4127, 'Year_of_Release'] = 2006.0

In [30]:
# https://gamefaqs.gamespot.com/gba/920485-game-boy-advance-video-the-fairly-oddparents-volume-1
video_games.loc[4508]

Name               The Fairly Odd Parents: Game Boy Advance Video...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.31
EU_Sales                                                        0.11
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.43
Developer                                                        NaN
Rating                                                             0
Name: 4508, dtype: object

In [31]:
# Corrigindo informações da linha 4508
video_games.loc[4508, 'Publisher'] = 'Majesco'
video_games.loc[4508, 'Developer'] = 'Majesco'

In [32]:
# https://gamefaqs.gamespot.com/gba/920486-game-boy-advance-video-the-fairly-oddparents-volume-2/data
video_games.loc[4615]

Name               The Fairly Odd Parents: Game Boy Advance Video...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                         0.3
EU_Sales                                                        0.11
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.42
Developer                                                        NaN
Rating                                                             0
Name: 4615, dtype: object

In [33]:
# Corrigindo informações da linha 4615
video_games.loc[4615, 'Publisher'] = 'Majesco'
video_games.loc[4615, 'Developer'] = 'Majesco'

In [34]:
# https://www.ebay.com/p/54825746
# https://en.wikipedia.org/wiki/Dragon_Ball_Z:_Budokai_Tenkaichi
video_games.loc[5294]

Name               Dragon Ball Z: Budokai Tenkaichi 2 (JP sales)
Platform                                                     Wii
Year_of_Release                                              NaN
Genre                                                     Action
Publisher                                                    NaN
NA_Sales                                                    0.15
EU_Sales                                                    0.05
JP_Sales                                                    0.14
Other_Sales                                                 0.01
Global_Sales                                                0.35
Developer                                                    NaN
Rating                                                         0
Name: 5294, dtype: object

In [35]:
# Corrigindo informações da linha 5294
video_games.loc[5294,'Publisher'] = 'Atari Inc.'
video_games.loc[5294,'Developer'] = 'Spike'
video_games.loc[5294,'Year_of_Release'] = 2006.0

In [36]:
# https://gamefaqs.gamespot.com/gba/629322-game-boy-advance-video-cartoon-network-collection-platinum
video_games.loc[5633]

Name               Cartoon Network Collection: Game Boy Advance V...
Platform                                                         GBA
Year_of_Release                                               2005.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.23
EU_Sales                                                        0.08
JP_Sales                                                         0.0
Other_Sales                                                     0.01
Global_Sales                                                    0.32
Developer                                                        NaN
Rating                                                             0
Name: 5633, dtype: object

In [37]:
# Corrigindo informações da linha 5633
video_games.loc[5633, 'Publisher'] = 'Majesco'
video_games.loc[5633, 'Developer'] = 'Majesco'

In [38]:
# https://zelda.fandom.com/wiki/The_Legend_of_Zelda:_The_Minish_Cap
video_games.loc[6256]

Name               The Legend of Zelda: The Minish Cap(weekly JP ...
Platform                                                         GBA
Year_of_Release                                                  NaN
Genre                                                         Action
Publisher                                                        NaN
NA_Sales                                                         0.0
EU_Sales                                                         0.0
JP_Sales                                                        0.27
Other_Sales                                                     0.01
Global_Sales                                                    0.27
Developer                                                        NaN
Rating                                                             0
Name: 6256, dtype: object

In [39]:
# Corrigindo informações da linha 6256
video_games.loc[6256, 'Publisher'] = 'Nintendo'
video_games.loc[6256, 'Developer'] = 'Capcom'
video_games.loc[6256, 'Year_of_Release'] = 2004.0

In [40]:
# https://sonic.fandom.com/wiki/Sonic_X:_A_Super_Sonic_Hero_(Game_Boy_Advance)
video_games.loc[6415]

Name               Sonic X: Game Boy Advance Video Volume 1
Platform                                                GBA
Year_of_Release                                      2004.0
Genre                                                  Misc
Publisher                                               NaN
NA_Sales                                               0.19
EU_Sales                                               0.07
JP_Sales                                                0.0
Other_Sales                                             0.0
Global_Sales                                           0.27
Developer                                               NaN
Rating                                                    0
Name: 6415, dtype: object

In [41]:
# Corrigindo informações da linha 6415
video_games.loc[6415, 'Publisher'] = 'Majesco'
video_games.loc[6415, 'Developer'] = 'Majesco'

In [42]:
# https://gba-video.fandom.com/wiki/Dora_the_Explorer_Volume_1
video_games.loc[6532]

Name               Dora the Explorer: Game Boy Advance Video Volu...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.18
EU_Sales                                                        0.07
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.26
Developer                                                        NaN
Rating                                                             0
Name: 6532, dtype: object

In [43]:
# Corrigindo informações da linha 6532
video_games.loc[6532, 'Publisher'] = 'Nintendo'
video_games.loc[6532, 'Developer'] = 'Majesco'

In [44]:
# https://gba-video.fandom.com/wiki/Cartoon_Network_Collection_Volume_1
video_games.loc[6626]

Name               Cartoon Network Collection: Game Boy Advance V...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.18
EU_Sales                                                        0.07
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.25
Developer                                                        NaN
Rating                                                             0
Name: 6626, dtype: object

In [45]:
# Corrigindo informações da linha 6626
video_games.loc[6626, 'Publisher'] = 'Nintendo'
video_games.loc[6626, 'Developer'] = 'Majesco'

In [46]:
# https://rugrats.fandom.com/wiki/All_Grown_Up!:_Volume_1
video_games.loc[6831]

Name               All Grown Up!: Game Boy Advance Video Volume 1
Platform                                                      GBA
Year_of_Release                                            2004.0
Genre                                                        Misc
Publisher                                                     NaN
NA_Sales                                                     0.17
EU_Sales                                                     0.06
JP_Sales                                                      0.0
Other_Sales                                                   0.0
Global_Sales                                                 0.24
Developer                                                     NaN
Rating                                                          0
Name: 6831, dtype: object

In [47]:
# Corrigindo informações da linha 6831
video_games.loc[6831, 'Publisher'] = 'Nintendo'
video_games.loc[6831, 'Developer'] = 'Majesco'

In [48]:
# https://gamefaqs.gamespot.com/gba/924902-game-boy-advance-video-nicktoons-collection-volume-2
video_games.loc[7189]

Name               Nicktoons Collection: Game Boy Advance Video V...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.16
EU_Sales                                                        0.06
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.22
Developer                                                        NaN
Rating                                                             0
Name: 7189, dtype: object

In [49]:
# Corrigindo informações da linha 7189
video_games.loc[7189, 'Publisher'] = 'Majesco'
video_games.loc[7189, 'Developer'] = 'Majesco'

In [50]:
# https://en.wikipedia.org/wiki/Yu_Yu_Hakusho:_Dark_Tournament
video_games.loc[7333]

Name               Yu Yu Hakusho: Dark Tournament
Platform                                      PS2
Year_of_Release                               NaN
Genre                                    Fighting
Publisher                                     NaN
NA_Sales                                      0.1
EU_Sales                                     0.08
JP_Sales                                      0.0
Other_Sales                                  0.03
Global_Sales                                 0.21
Developer                         Digital Fiction
Rating                                          2
Name: 7333, dtype: object

In [51]:
# Corrigindo informações da linha 7333
video_games.loc[7333, 'Publisher'] = 'Atari'
video_games.loc[7333, 'Developer'] = 'Digital Fiction'
video_games.loc[7333, 'Year_of_Release'] = 2004.0

In [52]:
# https://gba-video.fandom.com/wiki/SpongeBob_SquarePants_Volume_3
video_games.loc[7477]

Name               SpongeBob SquarePants: Game Boy Advance Video ...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.15
EU_Sales                                                        0.05
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.21
Developer                                                        NaN
Rating                                                             0
Name: 7477, dtype: object

In [53]:
# Corrigindo informações da linha 7477
video_games.loc[7477, 'Publisher'] = 'Nintendo'
video_games.loc[7477, 'Developer'] = 'Majesco'

In [54]:
# https://gamefaqs.gamespot.com/snes/588788-thomas-the-tank-engine-and-friends/boxes/39058
video_games.loc[7930]

Name               Thomas the Tank Engine & Friends
Platform                                        GBA
Year_of_Release                              2004.0
Genre                                     Adventure
Publisher                                       NaN
NA_Sales                                       0.13
EU_Sales                                       0.05
JP_Sales                                        0.0
Other_Sales                                     0.0
Global_Sales                                   0.19
Developer                                       NaN
Rating                                            0
Name: 7930, dtype: object

In [55]:
# Corrigindo informações da linha 7930
video_games.loc[7930, 'Publisher'] = 'THQ'
video_games.loc[7930, 'Developer'] = 'Software Creations'

In [56]:
# https://gba-video.fandom.com/wiki/Dragon_Ball_GT_Volume_1
video_games.loc[8317]

Name               Dragon Ball GT: Game Boy Advance Video Volume 1
Platform                                                       GBA
Year_of_Release                                             2004.0
Genre                                                         Misc
Publisher                                                      NaN
NA_Sales                                                      0.12
EU_Sales                                                      0.05
JP_Sales                                                       0.0
Other_Sales                                                    0.0
Global_Sales                                                  0.17
Developer                                                      NaN
Rating                                                           0
Name: 8317, dtype: object

In [57]:
# Corrigindo informações da linha 8317
video_games.loc[8317, 'Publisher'] = 'Nintendo'
video_games.loc[8317, 'Developer'] = 'Majesco'

In [58]:
# https://gamefaqs.gamespot.com/gba/920483-game-boy-advance-video-codename-kids-next-door-volume-1
video_games.loc[8328]

Name               Codename: Kids Next Door: Game Boy Advance Vid...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.12
EU_Sales                                                        0.05
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.17
Developer                                                        NaN
Rating                                                             0
Name: 8328, dtype: object

In [59]:
# Corrigindo informações da linha 8328
video_games.loc[8328, 'Publisher'] = 'Majesco'
video_games.loc[8328, 'Developer'] = 'Majesco'

In [60]:
# https://gba-video.fandom.com/wiki/Teenage_Mutant_Ninja_Turtles_Volume_1
video_games.loc[8358]

Name               Teenage Mutant Ninja Turtles: Game Boy Advance...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.12
EU_Sales                                                        0.04
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.17
Developer                                                        NaN
Rating                                                             0
Name: 8358, dtype: object

In [61]:
# Corrigindo informações da linha 8358
video_games.loc[8358, 'Publisher'] = 'Nintendo'
video_games.loc[8358, 'Developer'] = 'Majesco'

In [62]:
# https://pt.wikipedia.org/wiki/Stronghold_3
video_games.loc[8494]

Name                  Stronghold 3
Platform                        PC
Year_of_Release             2011.0
Genre                     Strategy
Publisher                      NaN
NA_Sales                      0.06
EU_Sales                       0.1
JP_Sales                       0.0
Other_Sales                    0.0
Global_Sales                  0.16
Developer          FireFly Studios
Rating                           2
Name: 8494, dtype: object

In [63]:
# Corrigindo informações da linha 8494
video_games.loc[8494, 'Publisher'] = 'SouthPeak Games'

In [64]:
# https://gamefaqs.gamespot.com/gba/629324-game-boy-advance-video-cartoon-network-collection-special
video_games.loc[8754]

Name               Cartoon Network Collection: Game Boy Advance V...
Platform                                                         GBA
Year_of_Release                                               2005.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.11
EU_Sales                                                        0.04
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.15
Developer                                                        NaN
Rating                                                             0
Name: 8754, dtype: object

In [65]:
# Corrigindo informações da linha 8754
video_games.loc[8754, 'Publisher'] = 'Majesco'
video_games.loc[8754, 'Developer'] = 'Majesco'

In [66]:
# https://gamefaqs.gamespot.com/gba/922010-game-boy-advance-video-pokemon-johto-photo-finish
video_games.loc[8837]

Name               Pokémon: Johto Photo Finish: Game Boy Advance ...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.11
EU_Sales                                                        0.04
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.15
Developer                                                        NaN
Rating                                                             0
Name: 8837, dtype: object

In [67]:
# Corrigindo informações da linha 8837
video_games.loc[8837, 'Publisher'] = 'Nintendo'
video_games.loc[8837, 'Developer'] = 'Nintendo'

In [68]:
# https://gamefaqs.gamespot.com/gba/921950-game-boy-advance-video-strawberry-shortcake-volume-1
video_games.loc[8883]

Name               Strawberry Shortcake: Game Boy Advance Video V...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.11
EU_Sales                                                        0.04
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.15
Developer                                                        NaN
Rating                                                             0
Name: 8883, dtype: object

In [69]:
# Corrigindo informações da linha 8883
video_games.loc[8883, 'Publisher'] = 'Majesco'
video_games.loc[8883, 'Developer'] = 'Majesco'

In [70]:
# https://www.greenmangaming.com/pt/games/farming-simulator-2011-pc/?srsltid=AfmBOoo_3I-Lrj_aSMB0mm4NPIo4F8vQrJlQlKpTxczhkKnnFESsQhIO
video_games.loc[9511]

Name               Farming Simulator 2011
Platform                               PC
Year_of_Release                    2010.0
Genre                          Simulation
Publisher                             NaN
NA_Sales                              0.0
EU_Sales                             0.13
JP_Sales                              0.0
Other_Sales                           0.0
Global_Sales                         0.13
Developer                 Giants Software
Rating                                  0
Name: 9511, dtype: object

In [71]:
# Corrigindo informações da linha 9511
video_games.loc[9511, 'Publisher'] = 'Giants Software'

In [72]:
# https://gamefaqs.gamespot.com/psp/647928-super-robot-taisen-og-saga-masou-kishin-ii-revelation-of
video_games.loc[9739]

Name               Super Robot Wars OG Saga: Masou Kishin II - Re...
Platform                                                         PSP
Year_of_Release                                                  NaN
Genre                                                       Strategy
Publisher                                                        NaN
NA_Sales                                                         0.0
EU_Sales                                                         0.0
JP_Sales                                                        0.12
Other_Sales                                                      0.0
Global_Sales                                                    0.12
Developer                                                        NaN
Rating                                                             0
Name: 9739, dtype: object

In [73]:
# Corrigindo informações da linha 9739
video_games.loc[9739, 'Publisher'] = 'Bandai Namco Games'
video_games.loc[9739, 'Developer'] = 'WinkySoft'
video_games.loc[9739, 'Year_of_Release'] = 2012.0

In [74]:
# https://gamefaqs.gamespot.com/gba/925330-game-boy-advance-video-disney-channel-collection-volume-1
video_games.loc[10378]

Name               Disney Channel Collection Vol. 1
Platform                                        GBA
Year_of_Release                              2004.0
Genre                                          Misc
Publisher                                       NaN
NA_Sales                                       0.08
EU_Sales                                       0.03
JP_Sales                                        0.0
Other_Sales                                     0.0
Global_Sales                                   0.11
Developer                                       NaN
Rating                                            0
Name: 10378, dtype: object

In [75]:
# Corrigindo informações da linha 10378
video_games.loc[10378, 'Publisher'] = 'Majesco'
video_games.loc[10378, 'Developer'] = 'Disney Interactive Studios'

In [76]:
# https://gamefaqs.gamespot.com/ds/933405-atsumare-power-pro-kun-no-ds-koushien
video_games.loc[10487]

Name               Atsumare! Power Pro Kun no DS Koushien
Platform                                               DS
Year_of_Release                                       NaN
Genre                                              Sports
Publisher                                             NaN
NA_Sales                                              0.0
EU_Sales                                              0.0
JP_Sales                                              0.1
Other_Sales                                           0.0
Global_Sales                                          0.1
Developer                                             NaN
Rating                                                  0
Name: 10487, dtype: object

In [77]:
# Corrigindo informações da linha 10487
video_games.loc[10487, 'Publisher'] = 'Konami'
video_games.loc[10487, 'Developer'] = 'Konami'
video_games.loc[10487, 'Year_of_Release'] = 2006.0

In [78]:
# https://gamefaqs.gamespot.com/ps/196539-action-man-operation-extreme
video_games.loc[11093]

Name               Action Man-Operation Extreme
Platform                                     PS
Year_of_Release                             NaN
Genre                                    Action
Publisher                                   NaN
NA_Sales                                   0.05
EU_Sales                                   0.03
JP_Sales                                    0.0
Other_Sales                                0.01
Global_Sales                               0.09
Developer                                   NaN
Rating                                        0
Name: 11093, dtype: object

In [79]:
# Corrigindo informações da linha 11093
video_games.loc[11093, 'Publisher'] = 'Hasbro Interactive'
video_games.loc[11093, 'Developer'] = 'Blitz Games'
video_games.loc[11093, 'Year_of_Release'] = 2000.0

In [80]:
# https://gamefaqs.gamespot.com/gba/924901-game-boy-advance-video-cartoon-network-collection-volume-2
video_games.loc[11546]

Name               Cartoon Network Collection: Game Boy Advance V...
Platform                                                         GBA
Year_of_Release                                               2004.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.06
EU_Sales                                                        0.02
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.08
Developer                                                        NaN
Rating                                                             0
Name: 11546, dtype: object

In [81]:
# Corrigindo informações da linha 11546
video_games.loc[11546, 'Publisher'] = 'Majesco'
video_games.loc[11546, 'Developer'] = 'Majesco'


In [82]:
# https://gamefaqs.gamespot.com/ds/934355-chou-soujuu-mecha-mg/reviews/170076
video_games.loc[12525]

Name               Chou Soujuu Mecha MG
Platform                             DS
Year_of_Release                     NaN
Genre                        Simulation
Publisher                           NaN
NA_Sales                            0.0
EU_Sales                            0.0
JP_Sales                           0.06
Other_Sales                         0.0
Global_Sales                       0.06
Developer                           NaN
Rating                                0
Name: 12525, dtype: object

In [83]:
# Corrigindo informações da linha 12525
video_games.loc[12525, 'Publisher'] = 'Nintendo'
video_games.loc[12525, 'Developer'] = 'Sandlot'
video_games.loc[12525, 'Year_of_Release'] = 2006.0

In [84]:
# https://en.wikipedia.org/wiki/Prinny:_Can_I_Really_Be_the_Hero%3F
video_games.loc[12569]

Name               Prinny: Can I Really Be The Hero? (US sales)
Platform                                                    PSP
Year_of_Release                                             NaN
Genre                                                    Action
Publisher                                                   NaN
NA_Sales                                                   0.06
EU_Sales                                                    0.0
JP_Sales                                                    0.0
Other_Sales                                                 0.0
Global_Sales                                               0.06
Developer                                                   NaN
Rating                                                        0
Name: 12569, dtype: object

In [85]:
# Corrigindo informações da linha 12569
video_games.loc[12569, 'Publisher'] = 'Nippon Ichi Software'
video_games.loc[12569, 'Developer'] = 'Nippon Ichi Software'
video_games.loc[12569, 'Year_of_Release'] = 2009.0

In [86]:
# https://en.wikipedia.org/wiki/Monster_Hunter_Frontier_G
video_games.loc[13318]

Name               Monster Hunter Frontier Online
Platform                                      PS3
Year_of_Release                               NaN
Genre                                Role-Playing
Publisher                                     NaN
NA_Sales                                      0.0
EU_Sales                                      0.0
JP_Sales                                     0.05
Other_Sales                                   0.0
Global_Sales                                 0.05
Developer                                     NaN
Rating                                          0
Name: 13318, dtype: object

In [87]:
# Corrigindo informações da linha 13318
video_games.loc[13318, 'Publisher'] = 'Capcom'
video_games.loc[13318, 'Developer'] = 'Capcom'
video_games.loc[13318, 'Year_of_Release'] = 2013.0

In [88]:
# https://gamefaqs.gamespot.com/ps/572662-blue-legend-of-water
video_games.loc[13712]

Name               B.L.U.E.: Legend of Water
Platform                                  PS
Year_of_Release                          NaN
Genre                              Adventure
Publisher                                NaN
NA_Sales                                 0.0
EU_Sales                                 0.0
JP_Sales                                0.04
Other_Sales                              0.0
Global_Sales                            0.04
Developer                                NaN
Rating                                     0
Name: 13712, dtype: object

In [89]:
# Corrigindo informações da linha 13712
video_games.loc[13712, 'Publisher'] = 'Hudson'
video_games.loc[13712, 'Developer'] = 'CAProduction'
video_games.loc[13712, 'Year_of_Release'] = 1998.0

In [90]:
# https://gamefaqs.gamespot.com/ds/939478-bikkuriman-daijiten
video_games.loc[14352]

Name               Bikkuriman Daijiten
Platform                            DS
Year_of_Release                    NaN
Genre                             Misc
Publisher                          NaN
NA_Sales                           0.0
EU_Sales                           0.0
JP_Sales                          0.03
Other_Sales                        0.0
Global_Sales                      0.03
Developer                          NaN
Rating                               0
Name: 14352, dtype: object

In [91]:
# Corrigindo informações da linha 14352
video_games.loc[14352, 'Publisher'] = "3 O'Clock"
video_games.loc[14352, 'Developer'] = "3 O'Clock"
video_games.loc[14352, 'Year_of_Release'] = 2007.0

In [92]:
# https://pt.wikipedia.org/wiki/World_of_Tanks
video_games.loc[14687]

Name                       World of Tanks
Platform                             X360
Year_of_Release                       NaN
Genre                             Shooter
Publisher                             NaN
NA_Sales                              0.0
EU_Sales                             0.03
JP_Sales                              0.0
Other_Sales                           0.0
Global_Sales                         0.03
Developer          Microsoft Game Studios
Rating                                  0
Name: 14687, dtype: object

In [93]:
# Corrigindo informações da linha 14687
video_games.loc[14687, 'Publisher'] = 'Microsoft Game Studios'
video_games.loc[14687, 'Year_of_Release'] = 2014.0

In [94]:
# https://gamefaqs.gamespot.com/pc/981331-uk-truck-simulator/data
video_games.loc[14772]

Name               UK Truck Simulator
Platform                           PC
Year_of_Release                2010.0
Genre                      Simulation
Publisher                         NaN
NA_Sales                          0.0
EU_Sales                         0.03
JP_Sales                          0.0
Other_Sales                       0.0
Global_Sales                     0.03
Developer                         NaN
Rating                              0
Name: 14772, dtype: object

In [95]:
# Corrigindo informações da linha 14772
video_games.loc[14772, 'Publisher'] = 'Excalibur Publishing'
video_games.loc[14772, 'Developer'] = 'SCS Software'

In [96]:
# https://gamefaqs.gamespot.com/ps3/637324-umineko-no-naku-koro-ni-san-shinjitsu-to-gensou-no
video_games.loc[15024]

Name               Umineko no Naku Koro ni San: Shinjitsu to Gens...
Platform                                                         PS3
Year_of_Release                                                  NaN
Genre                                                      Adventure
Publisher                                                        NaN
NA_Sales                                                         0.0
EU_Sales                                                         0.0
JP_Sales                                                        0.02
Other_Sales                                                      0.0
Global_Sales                                                    0.02
Developer                                                        NaN
Rating                                                             0
Name: 15024, dtype: object

In [97]:
# Corrigindo informações da linha 15024
video_games.loc[15024, 'Publisher'] = 'Alchemist'
video_games.loc[15024, 'Developer'] = '07th Expansion'
video_games.loc[15024, 'Year_of_Release'] = 2011.0

In [98]:
# https://en.wikipedia.org/wiki/Mario_Tennis_Open
video_games.loc[15340]

Name               Mario Tennis
Platform                    3DS
Year_of_Release             NaN
Genre                    Sports
Publisher                   NaN
NA_Sales                    0.0
EU_Sales                    0.0
JP_Sales                   0.02
Other_Sales                 0.0
Global_Sales               0.02
Developer                   NaN
Rating                        0
Name: 15340, dtype: object

In [99]:
# Corrigindo informações da linha 15340
video_games.loc[15340, 'Publisher'] = 'Nintendo'
video_games.loc[15340, 'Developer'] = 'Camelot Software Planning'
video_games.loc[15340, 'Year_of_Release'] = 2012.0

In [100]:
# https://gamefaqs.gamespot.com/gba/927237-game-boy-advance-video-nicktoons-collection-volume-3/data
video_games.loc[15417]

Name               Nicktoons Collection: Game Boy Advance Video V...
Platform                                                         GBA
Year_of_Release                                               2005.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.01
EU_Sales                                                        0.01
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.02
Developer                                                        NaN
Rating                                                             0
Name: 15417, dtype: object

In [101]:
# Corrigindo informações da linha 15417
video_games.loc[15417, 'Publisher'] = 'Majesco'
video_games.loc[15417, 'Developer'] = 'Majesco'

In [102]:
# https://www.gamersgate.com/product/demolition-company-gold-edition/?srsltid=AfmBOorVKxhnFfTw8dqy8AgSBeBmXFbCcSb1z2EL2m86qBhuEf9cRo8u
video_games.loc[15442]

Name               Demolition Company: Gold Edition
Platform                                         PC
Year_of_Release                              2011.0
Genre                                    Simulation
Publisher                                       NaN
NA_Sales                                        0.0
EU_Sales                                       0.02
JP_Sales                                        0.0
Other_Sales                                     0.0
Global_Sales                                   0.02
Developer                           Giants Software
Rating                                            0
Name: 15442, dtype: object

In [103]:
# Corrigindo informações da linha 15442
video_games.loc[15442, 'Publisher'] = 'GIANTS Software GmbH'

In [104]:
# https://otomeorganization.fandom.com/wiki/Moshi,_Kono_Sekai_ni_Kami-sama_ga_Iru_to_Suru_Naraba.
video_games.loc[15680]

Name               Moshi, Kono Sekai ni Kami-sama ga Iru to suru ...
Platform                                                         PSV
Year_of_Release                                               2016.0
Genre                                                      Adventure
Publisher                                                        NaN
NA_Sales                                                         0.0
EU_Sales                                                         0.0
JP_Sales                                                        0.02
Other_Sales                                                      0.0
Global_Sales                                                    0.02
Developer                                                        NaN
Rating                                                             0
Name: 15680, dtype: object

In [105]:
# Corrigindo informações da linha 15680
video_games.loc[15680, 'Publisher'] = 'Rejet'
video_games.loc[15680, 'Developer'] = 'Rejet'

In [106]:
# https://gamefaqs.gamespot.com/ds/973034-dream-dancer
video_games.loc[16019]

Name                                           Dream Dancer
Platform                                                 DS
Year_of_Release                                         NaN
Genre                                                  Misc
Publisher                                               NaN
NA_Sales                                               0.01
EU_Sales                                                0.0
JP_Sales                                                0.0
Other_Sales                                             0.0
Global_Sales                                           0.02
Developer          CyberPlanet Interactive Public Co., Ltd.
Rating                                                    0
Name: 16019, dtype: object

In [107]:
# Corrigindo informações da linha 16019
video_games.loc[16019, 'Publisher'] = 'Zoo Games'
video_games.loc[16019, 'Year_of_Release'] = 2009.0

In [108]:
# https://www.pcgamebenchmark.com/homeworld-remastered-collection-system-requirements
video_games.loc[16280]

Name               Homeworld Remastered Collection
Platform                                        PC
Year_of_Release                                NaN
Genre                                     Strategy
Publisher                                      NaN
NA_Sales                                       0.0
EU_Sales                                      0.01
JP_Sales                                       0.0
Other_Sales                                    0.0
Global_Sales                                  0.01
Developer                         Gearbox Software
Rating                                           3
Name: 16280, dtype: object

In [109]:
# Corrigindo informações da linha 16280
video_games.loc[16289, 'Publisher'] = 'Gearbox Software'
video_games.loc[16289, 'Year_of_Release'] = 2015.0

In [110]:
# https://gamefaqs.gamespot.com/psp/635168-akb1-48-idol-to-guam-de-koishitara/cheats
video_games.loc[16296]

Name               AKB1/48: Idol to Guam de Koishitara...
Platform                                             X360
Year_of_Release                                       NaN
Genre                                                Misc
Publisher                                             NaN
NA_Sales                                              0.0
EU_Sales                                              0.0
JP_Sales                                             0.01
Other_Sales                                           0.0
Global_Sales                                         0.01
Developer                                             NaN
Rating                                                  0
Name: 16296, dtype: object

In [111]:
# Corrigindo informações da linha 16296
video_games.loc[16296, 'Publisher'] = 'Bandai Namco Games'
video_games.loc[16296, 'Developer'] = 'Bandai Namco Games'
video_games.loc[16296, 'Year_of_Release'] = 2011.0

In [112]:
# https://gamefaqs.gamespot.com/gba/927238-game-boy-advance-video-super-robot-monkey-team-hyper-force
video_games.loc[16306]

Name               Super Robot Monkey Team: Game Boy Advance Vide...
Platform                                                         GBA
Year_of_Release                                               2005.0
Genre                                                           Misc
Publisher                                                        NaN
NA_Sales                                                        0.01
EU_Sales                                                         0.0
JP_Sales                                                         0.0
Other_Sales                                                      0.0
Global_Sales                                                    0.01
Developer                                                        NaN
Rating                                                             0
Name: 16306, dtype: object

In [113]:
# Corrigindo informações da linha 16306
video_games.loc[16306, 'Publisher'] = 'Majesco'
video_games.loc[16306, 'Developer'] = 'Majesco'

In [114]:
# https://en.wikipedia.org/wiki/Brothers_in_Arms:_Furious_4
video_games.loc[16332]

Name               Brothers in Arms: Furious 4
Platform                                  X360
Year_of_Release                            NaN
Genre                                  Shooter
Publisher                                  NaN
NA_Sales                                  0.01
EU_Sales                                   0.0
JP_Sales                                   0.0
Other_Sales                                0.0
Global_Sales                              0.01
Developer                     Gearbox Software
Rating                                       1
Name: 16332, dtype: object

In [115]:
# O Jogo foi cancelado
video_games = video_games.drop(16332)

In [116]:
# https://otome-games.fandom.com/wiki/Dance_with_Devils
video_games.loc[16339]

Name               Dance with Devils
Platform                         PSV
Year_of_Release               2016.0
Genre                         Action
Publisher                        NaN
NA_Sales                         0.0
EU_Sales                         0.0
JP_Sales                        0.01
Other_Sales                      0.0
Global_Sales                    0.01
Developer                        NaN
Rating                             0
Name: 16339, dtype: object

In [117]:
# Corrigindo informações da linha 16339
video_games.loc[16339, 'Publisher'] = 'Rejet'
video_games.loc[16339, 'Developer'] = 'Rejet'

In [118]:
# https://gamefaqs.gamespot.com/pc/621443-driving-simulator-2011
video_games.loc[16662]

Name               Driving Simulator 2011
Platform                               PC
Year_of_Release                    2011.0
Genre                              Racing
Publisher                             NaN
NA_Sales                              0.0
EU_Sales                             0.01
JP_Sales                              0.0
Other_Sales                           0.0
Global_Sales                         0.01
Developer                             NaN
Rating                                  0
Name: 16662, dtype: object

In [119]:
# Corrigindo informações da linha 16662
video_games.loc[16662, 'Publisher'] = 'Excalibur Publishing'
video_games.loc[16662, 'Developer'] = 'Excalibur Publishing'

In [120]:
# Apagando jogos que não encontramos informações
video_games = video_games.drop(2212)
video_games = video_games.drop(14128)

In [121]:
# Remove as linhas que contém os dados faltantes, por se tratar de uma pequena quantia em relação a base
video_games = video_games.dropna(subset=['Year_of_Release']).copy()
# Converter 'Year_of_Release' para numérico, forçando erros para NaN
video_games.loc[:, 'Year_of_Release'] = pd.to_numeric(video_games['Year_of_Release'], errors='coerce')

In [122]:
# Preencher valores nulos em 'Publisher' com uma categoria 'Unknown', pois o valor exato pode não ser crítico
video_games.fillna({'Publisher': 'Unknown'}, inplace=True)

In [123]:
# Preencher valores nulos em 'Developer' com uma categoria 'Unknown', pois o valor exato pode não ser crítico
video_games.fillna({'Developer': 'Unknown'}, inplace=True)

In [124]:
video_games.isnull().sum()

Name               0
Platform           0
Year_of_Release    0
Genre              0
Publisher          0
NA_Sales           0
EU_Sales           0
JP_Sales           0
Other_Sales        0
Global_Sales       0
Developer          0
Rating             0
dtype: int64

In [125]:
video_games.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16466 entries, 0 to 16718
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16466 non-null  object 
 1   Platform         16466 non-null  object 
 2   Year_of_Release  16466 non-null  float64
 3   Genre            16466 non-null  object 
 4   Publisher        16466 non-null  object 
 5   NA_Sales         16466 non-null  float64
 6   EU_Sales         16466 non-null  float64
 7   JP_Sales         16466 non-null  float64
 8   Other_Sales      16466 non-null  float64
 9   Global_Sales     16466 non-null  float64
 10  Developer        16466 non-null  object 
 11  Rating           16466 non-null  int64  
dtypes: float64(6), int64(1), object(5)
memory usage: 1.6+ MB


In [126]:
video_games.describe()

Year_of_Release      NA_Sales      EU_Sales      JP_Sales  \
count     16466.000000  16466.000000  16466.000000  16466.000000   
mean       2006.489979      0.263802      0.145856      0.078438   
std           5.876721      0.817945      0.506462      0.310904   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2020.000000     41.360000     28.960000     10.220000   

        Other_Sales  Global_Sales        Rating  
count  16466.000000  16466.000000  16466.000000  
mean       0.047565      0.535917      0.704178  
std        0.187908      1.557804      1.041911  
min        0.000000      0.010000      0.000000  
25%        0.000000      0.060000      0.000000  
50%        0.010000      0.170000      0.000000  
75%        0.030000      0.470000      2.000000  
max       10.570000     82.530000      7.000000

In [127]:
video_games.head(60)

Name Platform  Year_of_Release  \
0                                     Wii Sports      Wii           2006.0   
1                              Super Mario Bros.      NES           1985.0   
2                                 Mario Kart Wii      Wii           2008.0   
3                              Wii Sports Resort      Wii           2009.0   
4                       Pokemon Red/Pokemon Blue       GB           1996.0   
5                                         Tetris       GB           1989.0   
6                          New Super Mario Bros.       DS           2006.0   
7                                       Wii Play      Wii           2006.0   
8                      New Super Mario Bros. Wii      Wii           2009.0   
9                                      Duck Hunt      NES           1984.0   
10                                    Nintendogs       DS           2005.0   
11                                 Mario Kart DS       DS           2005.0   
12                   Pokemon Gold/Pokemon Silver       GB           1999.0   
13                                       Wii Fit      Wii           2007.0   
14                            Kinect Adventures!     X360           2010.0   
15                                  Wii Fit Plus      Wii           2009.0   
16                            Grand Theft Auto V      PS3           2013.0   
17                 Grand Theft Auto: San Andreas      PS2           2004.0   
18                             Super Mario World     SNES           1990.0   
19  Brain Age: Train Your Brain in Minutes a Day       DS           2005.0   
20                 Pokemon Diamond/Pokemon Pearl       DS           2006.0   
21                              Super Mario Land       GB           1989.0   
22                           Super Mario Bros. 3      NES           1988.0   
23                            Grand Theft Auto V     X360           2013.0   
24                   Grand Theft Auto: Vice City      PS2           2002.0   
25                 Pokemon Ruby/Pokemon Sapphire      GBA           2002.0   
26   Brain Age 2: More Training in Minutes a Day       DS           2005.0   
27                   Pokemon Black/Pokemon White       DS           2010.0   
28                        Gran Turismo 3: A-Spec      PS2           2001.0   
29                Call of Duty: Modern Warfare 3     X360           2011.0   
30       Pokémon Yellow: Special Pikachu Edition       GB           1998.0   
31                     Call of Duty: Black Ops 3      PS4           2015.0   
32                       Call of Duty: Black Ops     X360           2010.0   
33                           Pokemon X/Pokemon Y      3DS           2013.0   
34                    Call of Duty: Black Ops II      PS3           2012.0   
35                    Call of Duty: Black Ops II     X360           2012.0   
36                Call of Duty: Modern Warfare 2     X360           2009.0   
37                Call of Duty: Modern Warfare 3      PS3           2011.0   
38                          Grand Theft Auto III      PS2           2001.0   
39                       Super Smash Bros. Brawl      Wii           2008.0   
40                                  Mario Kart 7      3DS           2011.0   
41                       Call of Duty: Black Ops      PS3           2010.0   
42                            Grand Theft Auto V      PS4           2014.0   
43                   Animal Crossing: Wild World       DS           2005.0   
44                                        Halo 3     X360           2007.0   
45                                Super Mario 64      N64           1996.0   
46          Pokemon HeartGold/Pokemon SoulSilver       DS           2009.0   
47     Pokemon Omega Ruby/Pokemon Alpha Sapphire      3DS           2014.0   
48                                Gran Turismo 4      PS2           2004.0   
49                            Super Mario Galaxy      Wii           2007.0   
50            Super Mario Land 2: 6 Golden Coins       GB         

In [128]:
video_games.to_csv('./data/Clear_Video_Games.csv' , index=False)